### PR TITLE
changes : Added more documentation and grammatically fixed some.

### PR DESCRIPTION
**Pull request by relaxxpls**
1. Corrected a few grammatical errors and typos in the docs/references/account/create-session-oauth2.md 
2. Added a general readme for all the OAuths.
3. Added a readme for Google OAuth 2.0 integration and documented its functions.

### CODE REVIEW

**Comment by eldadfux** [comment]
Hey @relaxxpls, please submit the OAuth tutorial in this PR. We can't accept a PR with a link to a tutorial that does not exists yet. Thanks!

**Comment by relaxxpls** [comment]
I've updated the pr adding the Google OAuth tutorial.

**Comment by relaxxpls** [comment]
@eldadfux could you checkout this pr?

**Comment by Kolkies** [comment]
Will look into this asap.

File: docs/oauth/OAuth-Google/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+```
+- Returns the array of all the details(ID, email, name) of the user whose access token is passed  
+
+### Note:
+#### Rate Limits
+This endpoint is limited to 50 requests in every 60 minutes per IP address.   
+We use rate limits to avoid service abuse by users and as a security practice.  
+[Learn more about rate limiting](https://appwrite.io/docs/rate-limits)
+
+Refrances:
```
Comment (Kolkies) [inline docs/oauth/OAuth-Google/README.md:101]: I guess this should be `References`

**Review by Kolkies — CHANGES_REQUESTED** [review]

File: docs/oauth/OAuth-Google/README.md (hunk 1 above)
Comment (relaxxpls) [reply docs/oauth/OAuth-Google/README.md:101]: My bad, I'll fix it in a second!

Comment (relaxxpls) [reply docs/oauth/OAuth-Google/README.md]: @Kolkies Done! Could you confirm?

**Comment by eldadfux** [comment]
@relaxxpls a few things to fix:
1. Please use semantic titling, title needs to be in a logical hierarchy (h1 title => #, h2 title => ## and so on)
2. Please spell Appwrite and not AppWrite

**Comment by relaxxpls** [comment]
> @relaxxpls a few things to fix:
> 
> 1. Please use semantic titling, title needs to be in a logical hierarchy (h1 title => #, h2 title => ## and so on)
> 2. Please spell Appwrite and not AppWrite

Will do soon!

**Comment by relaxxpls** [comment]
@eldadfux changes have been made! Could you confirm?

File: docs/oauth/OAuth-Google/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
+# Create Account Session through AppWrite with Google's OAuth 2.0!
+
+## Features :
```
Comment (Kolkies) [inline docs/oauth/OAuth-Google/README.md:3]:
```suggestion
## Features:
```

File: docs/oauth/OAuth-Google/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+sdk
+    .setEndpoint('https://[HOSTNAME_OR_IP]/v4/account/sessions/oauth2/google') // Your API Endpoint
+    .setProject('5df5acd0d48c2'); // Your application ID
+
+// Go to OAuth provider login page
+sdk.account.createOAuth2Session('google');
+```
+
+## Functions Appwrite provides :
```
Comment (Kolkies) [inline docs/oauth/OAuth-Google/README.md:64]:
```suggestion
## Functions Appwrite provides:
```

File: docs/oauth/README.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Tutorials for integrating Appwrite with the OAuth 2.0 API for easily managing new clients!  
+Here are various examples illustrated with the help of the Appwrite Community  
+
+## Choices AppWrite offers :
```
Comment (Kolkies) [inline docs/oauth/README.md:4]:
```suggestion
## Choices AppWrite offers:
```

**Comment by relaxxpls** [comment]
@eldadfux @Kolkies if its done, could you merge this pr?

**Pull request closed without merging**